Sources/RoutingKit/Parameters.swift

```diff
@@ -51,27 +52,25 @@ public struct Parameters: Sendable {
     /// Grabs the named parameter from the parameter bag, casting it to
     /// a `LosslessStringConvertible` type.
     ///
-    /// For example GET /posts/:post_id/comments/:comment_id
+    /// For example `GET /posts/:post_id/comments/:comment_id`
     /// would be fetched using:
     ///
     ///     let postID = parameters.get("post_id", as: Int.self)
     ///     let commentID = parameters.get("comment_id", as: Int.self)
     ///
-    public func get<T>(_ name: String, as type: T.Type = T.self) -> T?
-        where T: LosslessStringConvertible
-    {
+    public func get<T: LosslessStringConvertible>(_ name: String, as type: T.Type = T.self) -> T? {
         self.get(name).flatMap(T.init)
     }
     
     /// Adds a new parameter value to the bag.
     ///
-    /// - note: The value will be percent-decoded.
+    /// > Note: The value will be percent-decoded.
     ///
-    /// - parameters:
+    /// - Parameters:
     ///     - name: Unique parameter name
     ///     - value: Value (percent-encoded if necessary)
     public mutating func set(_ name: String, to value: String?) {
-        self.values[name] = value?.removingPercentEncoding
+        self.values[name] = value.map { $0.removingPercentEncoding ?? $0 }
```

Comment:
This will insert an unencoded value if the encoding fails right?

Reply:
Un**de**coded, but yes. This is the same fallback path that `getCatchall()` already had (which is in `setCatchall()` now). As noted in the PR description, it shouldn't be possible to get to that code path at all short of using RoutingKit directly, without Vapor, since the only reason `removingPercentEncoding` fails is for an invalid input (`%` not followed by two hex characters), in which case you can't de-percent-encode the input anyway.